storage/aptosdb/src/state_kv_db.rs

```diff
@@ -78,31 +79,85 @@ impl StateKvDb {
             }
         };
 
-        Ok(Self {
+        let state_kv_db = Self {
             state_kv_metadata_db,
-            _state_kv_db_shards,
-        })
+            state_kv_db_shards,
+        };
+
+        if let Some(overall_kv_commit_progress) = get_state_kv_commit_progress(&state_kv_db)? {
+            truncate_state_kv_db_shards(&state_kv_db, overall_kv_commit_progress, None)?;
```

Comment:
if we don't truncate, what would be the downside?

Reply:
Maybe no downside if we can guarantee we are going to write the same data immediately, but who knows if this will change.

Reply:
I don't want to assume "we will always write the same data".  The cost of this extra step is cheap, and it can make the code more robust.